src/controller/controller.ts

```diff
@@ -131,11 +131,20 @@ class Controller extends events.EventEmitter {
         // Check if we have to change the channel, only do this when adapter `resumed` because:
         // - `getNetworkParameters` might be return wrong info because it needs to propogate after backup restore
         // - If result is not `resumed` (`reset` or `restored`), the adapter should comission with the channel from `this.options.network`
-        if (startResult === 'resumed' && (await this.adapter.supportsChangeChannel())) {
+        if (startResult === 'resumed') {
             const netParams = await this.getNetworkParameters();
+            const configuredChannel = this.options.network.channelList[0];
+            const adapterChannel = netParams.channel;
 
-            if (this.options.network.channelList[0] !== netParams.channel) {
-                await this.changeChannel();
+            if (configuredChannel != adapterChannel) {
+                if (this.adapter.supportsChangeChannel()) {
```

Comment:
Maybe we can just use a throw in `changeChannel` if not supported instead of 2 functions?
This codepath is expected to not be reached if adapter does not support channel changing anyway (will have done the logic to change by whatever means before this is reached).
Old refs: https://github.com/Koenkk/zigbee-herdsman/pull/977#discussion_r1527181194

Reply:
Agree, fixed!

Reply:
Oh, I meant in `changeChannel` of adapter, to be able to remove `supportsChangeChannel` completely.
_Sidenote: Probably should review these (there's one for backup too), the ones that are absolutely necessary and only return true/false should probably be moved to a var in adapter class or something._